pkg/ibmvpc/parser.go

```diff
@@ -767,99 +814,145 @@ func getIKSnodesConfig(res *vpcmodel.CloudConfig,
 		if err != nil {
 			return err
 		}
+
 		if skipByVPC(subnet.vpc.ResourceUID) {
 			continue
 		}
-
+		vpcUID := subnet.vpc.ResourceUID
 		nodeObject := &IKSNode{
 			VPCResource: vpcmodel.VPCResource{ResourceName: "iks-node", ResourceUID: iksNode.ID, ResourceType: ResourceTypeIKSNode},
 			address:     iksNode.IPAddress,
 			subnet:      subnet,
 		}
-		res.Nodes = append(res.Nodes, nodeObject)
+		res[vpcUID].Nodes = append(res[vpcUID].Nodes, nodeObject)
 		// attach the node to the subnet
 		subnet.nodes = append(subnet.nodes, nodeObject)
 	}
 	return nil
 }
 
-func NewCloudConfig(rc *ResourcesContainer, vpcID string) (*vpcmodel.CloudConfig, error) {
-	res := &vpcmodel.CloudConfig{
+func NewEmptyVPCConfig() *vpcmodel.VPCConfig {
+	return &vpcmodel.VPCConfig{
 		Nodes:            []vpcmodel.Node{},
 		NodeSets:         []vpcmodel.NodeSet{},
 		FilterResources:  []vpcmodel.FilterTrafficResource{},
 		RoutingResources: []vpcmodel.RoutingResource{},
 		NameToResource:   map[string]vpcmodel.VPCResourceIntf{},
 		CloudName:        "IBM Cloud",
 	}
+}
 
+// VPCConfigsFromResources returns a map from VPC UID (string) to its corresponding VPCConfig object,
+// containing the parsed resources in the relevant model objects
+func VPCConfigsFromResources(rc *ResourcesContainer, vpcID string, debug bool) (map[string]*vpcmodel.VPCConfig, error) {
+	res := map[string]*vpcmodel.VPCConfig{} // map from VPC UID to its config
+	vpcsMap := map[string]*VPC{}            // map from VPC UID to its object
```

Comment:
Does it make sense to add to `VPCConfig` a pointer to the VPC resource (of type `VPCResourceIntf`)?
This way, we can maintain only one map (`res`) and get rid of `vpcsMap`.

Reply:
The `VPCConfig`  is defined in the level of the abstract model, and the `VPC` resource type is defined in the concrete `ibmvpc` pkg, where it is parsed with all other concrete resources.

Reply:
ok, will add as another item in the issue